src/main.ts

```diff
@@ -22,14 +22,23 @@ async function run(): Promise<void> {
 
     switch (command) {
       case 'preview':
-        await BumpPreview.run(cliParams);
+        await bump.Preview.run(cliParams);
         break;
       case 'dry-run':
-      case 'validate':
-        await BumpDeploy.run(cliParams.concat(deployCliParams).concat(['--dry-run']));
+      case 'validate': // DEPRECATED, kept for backward compatibility with old gem
+        await bump.Deploy.run(cliParams.concat(docCliParams).concat(['--dry-run']));
         break;
       case 'deploy':
-        await BumpDeploy.run(cliParams.concat(deployCliParams));
+        await bump.Deploy.run(cliParams.concat(docCliParams));
+        break;
+      case 'diff':
+        await bump.Diff.run(cliParams.concat(docCliParams)).then(
+          (version: bump.VersionResponse | undefined) => {
+            if (version) {
+              diff.run(version);
+            }
+          },
+        );
```

Comment:
Why do we need this `version: bump.VersionResponse | undefined`. Don't we have any way to have a simple call here as we do for validate & deploy?

Reply:
I'm sorry I don't understand your question.

Here we want to chain two promises and I find it clearer with the `then` method on promises.

We can do it sequentially by writing:
```javascript
const version = await bump.Diff.run(...);
if (version) {
  await diff.run(version)
}
```

But as both promises are meant to work together I prefered the `promise.then(callback)` syntax in this case:
```javascript
await bump.Diff.run(...).then(
  (version) => {
    if (version) {
      await diff.run(version)
    }
  }
)
```

If it's not clear for you, I can write it by sequentially `await`ing each promise. You tell me!

Reply:
I agree, my question wasn't clear. In fact, there was 2 questions:
* first one is, why do we need a promise here. Couldn't we have a simple call, like `bump.Diff.run(cliParams.concat(docCliParams))` (without a promise), that would do 100% o the job?
* second one is: in which case should we have no version here? I get that, during processing, the server will return an empty 202 response, but at this point in the code it should already have been handled by the command, am I wrong?

Reply:
>     * first one is, why do we need a promise here. Couldn't we have a simple call, like `bump.Diff.run(cliParams.concat(docCliParams))` (without a promise), that would do 100% o the job?

Because the function `bump.Diff.run` is an async (promise) function [defined in the CLI](https://github.com/bump-sh/cli/blob/3f12ef115967b3862122eb41d9d2b721f98df952/src/commands/diff.ts#L54), where as the `diff.run` function is a new function [defined in this PR](https://github.com/paulRbr/github-action/blob/05fb65d2df6d827d917fb6fe0907189a24216802/src/diff.ts#L15).

Both functions are `async` because we use async functions (mostly because we do IO operations - e.g. HTTP requests) inside them.

>     * second one is: in which case should we have no version here? I get that, during processing, the server will return an empty 202 response, but at this point in the code it should already have been handled by the command, am I wrong?

Again because the function signature of `bump.Diff.run`[defined in the CLI](https://github.com/bump-sh/cli/blob/3f12ef115967b3862122eb41d9d2b721f98df952/src/commands/diff.ts#L54) is:

```typescript
 async run(): Promise<VersionResponse | void>
```

We return an empty (`void`) result when the spec file has not changed (when the POST /versions return a 204).

Reply:
OK, that makes sense. I guess that the 2 `Diff.run` & `diff.run` confused me here, sorry for that. Maybe we could, at some point, think about returning a null version object when nothing has changed to avoid this kind of `if`?

For now, LGTM :+1:, thanks for the explanation!

Reply:
You are right, we could improve the CLI `Diff` command signature to always return a `Version` object :+1: 

Overall, that's what is good with typing (and especially the fact that we force full signature typings): we have to be precise on variables, if they are defined or not, etc. All of this to avoid some “undefined | null” exceptions from occurring when we didn't expect it to :slightly_smiling_face: